redfish-core/lib/metric_definition.hpp

```diff
@@ -0,0 +1,404 @@
+
+    const auto* const it = std::find_if(
+        metricDefinitionMapping.begin(), metricDefinitionMapping.end(),
+        [&sensorObjectPath](const auto& item) {
+            return item.first == sensorObjectPath.parent_path().filename();
+        });
+
+    std::string metricDefinitionPath =
+        "/redfish/v1/TelemetryService/MetricDefinitions/";
+
+    if (it != metricDefinitionMapping.end())
+    {
+        return metricDefinitionPath + it->second;
+    }
+
+    return metricDefinitionPath + sensorObjectPath.filename();
+}
+
+static std::string& stringViewToStringRef(const std::string_view& sv)
+{
+    static std::string s;
+    s = std::string(sv);
+    return s;
+}
+
+template <class Callback>
+inline void mapRedfishUriToDbusPath(Callback&& callback)
+{
+    utils::getChassisNames([callback = std::forward<Callback>(callback)](
+                               boost::system::error_code ec,
+                               const std::vector<std::string>& chassisNames) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("getChassisNames error: {}", ec.value());
+            callback(ec, {});
+            return;
+        }
+
+        auto counter = std::make_shared<std::pair<
+            boost::container::flat_map<std::string, std::string>, size_t>>();
+
+        auto handleRetrieveUriToDbusMap =
+            [counter, callback = std::move(callback)](
+                const boost::beast::http::status status,
+                const std::map<std::string, std::string>& uriToDbus) {
+                if (status != boost::beast::http::status::ok)
+                {
+                    BMCWEB_LOG_ERROR("Failed to retrieve URI to dbus "
+                                     "sensors map with err {}",
+                                     static_cast<unsigned>(status));
+                    counter->second = 0U;
+                    callback(boost::system::errc::make_error_code(
+                                 boost::system::errc::io_error),
+                             {});
+                    return;
+                }
+
+                for (const auto& [key, value] : uriToDbus)
+                {
+                    counter->first[key] = value;
+                }
+
+                if (--counter->second == 0U)
+                {
+                    callback(boost::system::errc::make_error_code(
+                                 boost::system::errc::success),
+                             counter->first);
+                }
+            };
+
+        for (const std::string& chassisName : chassisNames)
+        {
+            ++counter->second;
+            retrieveUriToDbusMap(chassisName,
+                                 stringViewToStringRef(sensors::sensorsNodeStr),
+                                 handleRetrieveUriToDbusMap);
+        }
+    });
+}
+
+} // namespace telemetry
+
+inline void requestRoutesMetricDefinitionCollection(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/TelemetryService/MetricDefinitions/")
```

Comment:
The new way to do this is these ROUTEs sit below and then a function for each route
e.g. https://github.com/openbmc/bmcweb/blob/master/redfish-core/lib/power_supply.hpp#L594

It would make this more readable